src/services/create-profile-form-service.ts

```diff
@@ -47,17 +51,59 @@ export const validationSchema = Yup.object().shape({
   position: Yup.string().required('Position is required'),
   seniority: Yup.string().required('Seniority is required'),
   techStack: Yup.string().required('Tech stack is required'),
-  employment: Yup.array()
-    .of(Yup.string())
-    .min(1, 'Employment type is required'),
+  // employment: Yup.array().of(Yup.string().oneOf(['FULL_TIME', 'PART_TIME', 'CONTRACT'])).min(1, 'Employment type is required'),
 })
 
 export const useFormikInitialization = () => {
-  return useFormik<FormValues>({
-    initialValues,
-    validationSchema,
-    onSubmit: (values) => {
-      console.log(values)
-    },
-  })
+  const { data: session } = useSession()
+
+  const onSubmit = async (values: FormValues) => {
+    console.log('Form values:', values)
+    const payload: ProfilePayload = {
+      id: session?.user.id,
+      fullName: values.fullName,
+      email: session?.user.email,
+      linkedIn: values.linkedin,
+      bio: values.bio,
+      country: {
+        name: values.country,
+        openForRelocation: values.openToRelocationCountry,
+      },
+      city: {
+        name: values.city,
+        openForRelocation: values.openToRelocationCity,
+      },
+      remoteOnly: values.remoteOnly,
+      position: values.position,
+      seniority: values.seniority,
+      techStack: values.techStack.split(',').map((s) => s.trim()),
+      employmentType: values.employment,
+      isPublished: values.isPublished,
+    }
+
+    const response = await fetch('/api/profiles', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    })
+
+    if (response.ok) {
+      console.log('Profile created successfully')
+    } else {
+      console.log('Failed to create profile.')
+      const errorData = await response.json()
+      console.log('Error details:', errorData)
+    }
+  }
+
+  return {
+    formik: useFormik<FormValues>({
+      initialValues,
+      validationSchema,
+      onSubmit,
+    }),
+    onSubmit,
```

Comment:
**nitpick:** Why do you pass `onSubmit` if you can get it from `useFormik`? You pass it there anyway

Reply:
I'm unable to pass onSubmit to CreateProfileFormWrapper in any other way. If you find a solution, please let me know